source/dub/recipe/io.d

```diff
@@ -72,7 +73,27 @@ PackageRecipe parsePackageRecipe(string contents, string filename, string parent
 
 	ret.name = default_package_name;
 
-	if (filename.endsWith(".json")) parseJson(ret, parseJsonString(contents, filename), parent_name);
+	if (filename.endsWith(".json"))
+	{
+		ret = parseConfigString!PackageRecipe(contents, filename, StrictMode.Warn);
+		version (all)
+		{
+			import std.stdio;
+
+			PackageRecipe jsonret;
+			parseJson(jsonret, parseJsonString(contents, filename), parent_name);
+			if (ret != jsonret)
+			{
+				writeln("Content of JSON and YAML parsing differ for file: ", filename);
+				writeln("-------------------------------------------------------------------");
+				writeln("JSON (excepted): ", jsonret);
+				writeln("-------------------------------------------------------------------");
+				writeln("YAML (actual  ): ", ret);
+				writeln("========================================");
```

Comment:
these seem like debug output. If they stay in DUB for other users to report issues with the new parsing method, this should write to stderr.

Reply:
This is currently here while the PR is in draft. If people want to experiment with it, they will see if it works. That's also why the json parsed struct is returned if an error is detected. I plan to run this on every packages in the dub repository until everything can be parsed.